R/render_site.R

```diff
@@ -220,19 +220,39 @@ clean_site <- function(input = ".", preview = FALSE, quiet = FALSE,
   # get the files to be cleaned
   files <- generator$clean()
 
+
+  if (length(files) == 0) {
+    if (preview || !quiet) cat("Nothing to removed. All clean !\n")
+    return(invisible(NULL))
+  }
+
   # if it's just a preview then return the files, otherwise
   # actually remove the files
-  if (preview)
-    files
-  else {
+  if (preview) {
+    cat("These files and folders can probably be removed:\n",
+        paste0("* ", mark_dirs(files, input)),
+        "\nUse `rmarkdown::clean_site(preview = FALSE)` to remove them.",
+        sep = "\n")
+  } else {
     if (!quiet) {
-      cat("Removing files: \n")
-      cat(paste0(" ", files), sep = "\n")
+      cat("Removing files: \n",
+          paste0("* ", mark_dirs(files, input)),
+          sep = "\n")
     }
     unlink(file.path(input, files), recursive = TRUE)
   }
 }
 
+# TODO: Move to xfun - bookdown use it too.
+mark_dirs <- function(files, dir) {
+  if (!missing(dir)) {
+    owd = setwd(dir); on.exit(setwd(owd))
+  }
+  i = file_test("-d", files) & !grepl("/$", files)
+  files[i] = paste0(files[i], "/")
+  files
+}
```

Comment:
After some tests, I am not so sure the `dir` argument is useful because `files <- generator$clean()` does not pass the `input` directory to the function. 

That means I don't think `clean_site` can really be used with a directory different than the current working dir. 

Try bookdown-demo. After a first render, you won't get the same result between this
```
# Render book
xfun::in_dir("bookdown-demo/", bookdown::render_book("index.Rmd"))
# No file to remove ? 
rmarkdown::clean_site("bookdown-demo/", preview = TRUE)

# It seems there is
xfun::in_dir("bookdown-demo/", bookdown::clean_book())
```

Reply:
I removed the dir argument as discussed 33ef30f

We just need to remember that `clean_site()` only work within working directory really